tests/Unit/ApparentHorizons/Test_ApparentHorizonFinder.cpp

```diff
@@ -0,0 +1,371 @@
+};
+
+template <typename PostHorizonFindCallback>
+struct MockMetavariables {
+  struct AhA {
+    using compute_items_on_source = tmpl::list<
+        ah::Tags::InverseSpatialMetricCompute<3, Frame::Inertial>,
+        ah::Tags::ExtrinsicCurvatureCompute<3, Frame::Inertial>,
+        ah::Tags::SpatialChristoffelSecondKindCompute<3, Frame::Inertial>>;
+    using vars_to_interpolate_to_target =
+        tmpl::list<gr::Tags::InverseSpatialMetric<3, Frame::Inertial>,
+                   gr::Tags::ExtrinsicCurvature<3, Frame::Inertial>,
+                   gr::Tags::SpatialChristoffelSecondKind<3, Frame::Inertial>>;
+    using compute_items_on_target = tmpl::list<>;
+    using compute_target_points =
+        intrp::Actions::ApparentHorizon<AhA, ::Frame::Inertial>;
+    using post_interpolation_callback =
+        intrp::callbacks::FindApparentHorizon<AhA>;
+    using post_horizon_find_callback = PostHorizonFindCallback;
+    // This `type` is so this tag can be used to read options.
+    using type = typename compute_target_points::options_type;
+  };
+  using interpolator_source_vars =
+      tmpl::list<gr::Tags::SpacetimeMetric<3, Frame::Inertial>,
+                 GeneralizedHarmonic::Tags::Pi<3, Frame::Inertial>,
+                 GeneralizedHarmonic::Tags::Phi<3, Frame::Inertial>>;
+  using interpolation_target_tags = tmpl::list<AhA>;
+  using temporal_id = Time;
+  using domain_frame = Frame::Inertial;
+  static constexpr size_t domain_dim = 3;
+  using component_list =
+      tmpl::list<mock_interpolation_target<MockMetavariables, AhA>,
+                 mock_interpolator<MockMetavariables>>;
+  using const_global_cache_tag_list = tmpl::list<>;
+  enum class Phase { Initialize, Exit };
+};
+
+template <typename PostHorizonFindCallback>
+void test_apparent_horizon(const gsl::not_null<size_t*> test_horizon_called,
+                           const size_t l_max,
+                           const size_t grid_points_each_dimension,
+                           const double mass,
+                           const std::array<double, 3>& dimensionless_spin) {
+  using metavars = MockMetavariables<PostHorizonFindCallback>;
+  using MockRuntimeSystem = ActionTesting::MockRuntimeSystem<metavars>;
+  using mock_target_a =
+      mock_interpolation_target<metavars, typename metavars::AhA>;
+  using TupleOfMockDistributedObjects =
+      typename MockRuntimeSystem::TupleOfMockDistributedObjects;
+  TupleOfMockDistributedObjects dist_objects{};
+  using MockDistributedObjectsTagTargetA =
+      typename MockRuntimeSystem::template MockDistributedObjectsTag<
+          mock_target_a>;
+  using MockDistributedObjectsTagInterpolator =
+      typename MockRuntimeSystem::template MockDistributedObjectsTag<
+          mock_interpolator<metavars>>;
+
+  // The mocking framework requires an array index to be passed to
+  // many functions, even if the components are singletons.
+  constexpr size_t fake_array_index = 0_st;
```

Comment:
Singletons don't normally have ids, so I think `singleton_id` is misleading because it implies that it has some meaning.  It is only because of the mocking framework that you need to pass this extra argument, which has no meaning except in the tests, so I like `fake_array_index` better.

Reply:
I agree with Mark.

Reply:
ok that’s fine with me